WordPress/Classes/ViewRelated/Post/PostEditor+Publish.swift

```diff
@@ -192,19 +192,20 @@ extension PostEditor where Self: UIViewController {
             showPostHasChangesAlert()
         } else {
             editorSession.end(outcome: .cancel)
-            discardChangesAndUpdateGUI()
+            discardUnsavedChangesAndUpdateGUI()
         }
     }
 
-    func discardChangesAndUpdateGUI() {
-        discardChanges()
-
-        dismissOrPopView(didSave: false)
+    func discardUnsavedChangesAndUpdateGUI() {
+        let postDeleted = discardChanges()
```

Comment:
I wasn't supper happy about this, since it's not obvious that a method named `discardChanges` would return whether a post was deleted or not, and then `didSave == !postDeleted` seems like a non-obvious assumption. However I kept following the code and the whole thing is quite confusing, so I'll 🙈 for now and hope that having autosaves eventually allows us to simplify all this logic.

Reply:
I didn't like that either, but yes, the whole thing is a mess, we need to rewrite it.

Reply:
I agree.  I think it would be nice to try to think of ways in which we can incrementally improve this.